btcec/schnorr/musig2/nonces.go

```diff
@@ -0,0 +1,209 @@
+// Copyright 2013-2016 The btcsuite developers
+// Copyright (c) 2015-2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package musig2
+
+import (
+	"crypto/rand"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+)
+
+const (
+	// PubNonceSize is the size of the public nonces. Each public nonce is
+	// serialized the full compressed encoding, which uses 32 bytes for each
+	// nonce.
+	PubNonceSize = 66
+
+	// SecNonceSize is the size of the secret nonces for musig2. The secret
+	// nonces are the corresponding private keys to the public nonce points.
+	SecNonceSize = 64
+)
+
+// Nonces holds the public and secret nonces required for musig2.
+//
+// TODO(roasbeef): methods on this to help w/ parsing, etc?
+type Nonces struct {
+	// PubNonce holds the two 33-byte compressed encoded points that serve
+	// as the public set of nonces.
+	PubNonce [PubNonceSize]byte
+
+	// SecNonce holds the two 32-byte scalar values that are the private
+	// keys to the two public nonces.
+	SecNonce [SecNonceSize]byte
+}
+
+// secNonceToPubNonce takes our two secrete nonces, and produces their two
+// corresponding EC points, serialized in compressed format.
+func secNonceToPubNonce(secNonce [SecNonceSize]byte) [PubNonceSize]byte {
+	var k1Mod, k2Mod btcec.ModNScalar
+	k1Mod.SetByteSlice(secNonce[:])
```

Comment:
This makes an assumption about the implementation of the underlying `SetByteSlice` truncating anything after 32 bytes. Should we make it more explicit though by using `secNonce[:btcec.PrivKeyBytesLen]`?